storage/innobase/xtrabackup/src/xbcrypt_read.c

```diff
@@ -152,25 +157,14 @@ xb_crypt_read_chunk(xb_rcrypt_t *crypt, void **buf, size_t *olen, size_t *elen,
 	}
 
 	if (*ivlen > crypt->ivbufsize) {
-		if (crypt->ivbuffer) {
-			crypt->ivbuffer = my_realloc(crypt->ivbuffer, *ivlen,
-						   MYF(MY_WME));
-			if (crypt->ivbuffer == NULL) {
-				msg("%s:%s: failed to increase iv buffer to "
-				    "%llu bytes.\n", my_progname, __FUNCTION__,
-				    (ulonglong)*ivlen);
-				result = XB_CRYPT_READ_ERROR;
-				goto err;
-			}
-		} else {
-			crypt->ivbuffer = my_malloc(*ivlen, MYF(MY_WME));
-			if (crypt->ivbuffer == NULL) {
-				msg("%s:%s: failed to allocate iv buffer of "
-				    "%llu bytes.\n", my_progname, __FUNCTION__,
-				    (ulonglong)*ivlen);
-				result = XB_CRYPT_READ_ERROR;
-				goto err;
-			}
+		crypt->ivbuffer = my_realloc(crypt->ivbuffer, *ivlen,
+					     MYF(MY_WME | MY_ALLOW_ZERO_PTR));
+		if (crypt->ivbuffer == NULL) {
+			msg("%s:%s: failed to increase iv buffer to "
+			    "%llu bytes.\n", my_progname, __FUNCTION__,
+			    (ulonglong)*ivlen);
+			result = XB_CRYPT_READ_ERROR;
+			goto err;
```

Comment:
:) nice comic, I like to read it sometime as well. Actually `goto` is handy alternative to `try ... catch` blocks. It is widely used in the MySQL and xtrabackup code base. We don't use C++ exception handling. You may like it or not, but you have to accept it :)